skorch/helper.py

```diff
@@ -84,6 +89,46 @@ def __repr__(self):
     def shape(self):
         return (self._len,)
 
+    def copy(self):
+        return type(self)(**self)
+
+    def fromkeys(self, *args, **kwargs):
+        """fromkeys method makes no sense with SliceDict and is thus not
+        supported."""
+        raise TypeError("SliceDict does not support fromkeys.")
+
+    def __eq__(self, other):
+        if self.keys() != other.keys():
+            return False
+
+        for key, val in self.items():
+            val_other = other[key]
+
+            # torch tensors
+            if is_torch_data_type(val):
+                if not is_torch_data_type(val_other):
+                    return False
+                if not (val == val_other).all():
+                    return False
+                continue
+
+            # numpy arrays
+            if isinstance(val, np.ndarray):
+                if not isinstance(val_other, np.ndarray):
+                    return False
+                if not (val == val_other).all():
+                    return False
+                continue
```

Comment:
What about DataFrames? Can we use a more generic way for collections, e.g. `all(a == b)` for `array` and `DataFrame`s? (torch tensors don't work that way though)

Reply:
I thought about pandas `DataFrame`s but they probably don't matter. As I see it, they serve a similar purpose as a dict, namely to have named features (keys/columns) whose values may have different meanings. Therefore, I feel that you don't really need to mix both. In a way, a `DataFrame` is a dict with 1-dim arrays as values; `SliceDict` is a dict with n-dim arrays/tensors as values.